components/speedreader/speedreader_rewriter_unittest.cc

```diff
@@ -0,0 +1,78 @@
+
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include <string>
+
+#include "base/files/file_path.h"
+#include "base/files/file_util.h"
+#include "base/logging.h"
+#include "base/path_service.h"
+#include "base/threading/thread_restrictions.h"
+#include "brave/common/brave_paths.h"
+#include "brave/components/speedreader/rust/ffi/speedreader.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace speedreader {
+
+class SpeedreaderRewriterTest
+    : public ::testing::Test,
+      public ::testing::WithParamInterface<const char*> {
+ public:
+  SpeedreaderRewriterTest() {
+    brave::RegisterPathProvider();
+    base::PathService::Get(brave::DIR_TEST_DATA, &test_data_dir_);
+    test_data_dir_ = test_data_dir_.AppendASCII("speedreader/rewriter");
+  }
+
+  ~SpeedreaderRewriterTest() override = default;
+
+  std::string GetFileContent(const std::string& filename) {
+    std::string result;
+
+    const auto full_path = test_data_dir_.AppendASCII(filename);
+    /* void() for assert */
+    [&]() { ASSERT_TRUE(base::ReadFileToString(full_path, &result)); }();
+    return result;
+  }
+
+  std::string ProcessPage(const std::string& file_name) {
+    auto rewriter = speedreader_.MakeRewriter(
+        "https://test.com", RewriterType::RewriterReadability);
+    const auto file_content = GetFileContent(file_name);
+    rewriter->Write(file_content.data(), file_content.size());
+    rewriter->End();
+    return rewriter->GetOutput();
+  }
+
+  void CheckContent(const std::string& expected_content,
+                    const std::string& filename) {
+    EXPECT_EQ(GetFileContent(filename), expected_content) << expected_content;
```

Comment:
probably better to log the filename? expected content already gets printed if EXPECT fails

Reply:
Yes, but it will be printed with \n, and you can't just copy it without replacing \n with '\n'. This print is for convenience

Reply:
For these kind of tests I would create a minimal example that reproduces the problem and inline it in the rust file. For ex https://github.com/brave/brave-core/blob/b72b6eee5c6c23a5f33464c321b9eabff031d899/components/speedreader/rust/lib/src/readability/src/extractor.rs#L1086

PR is very high quality. Glad to see someone working on it.